### ISSUE TITLE
 Accordion buttons have incomplete accessible names (WCAG Level A failure)

### ISSUE THREAD

**Issue by maggiewachs** [swagger-api/swagger-ui#9353]
### Q&A (please complete the following information)
 - OS: all
 - Browser: all
 - Version: all
 - Method of installation: n/a
 - Swagger-UI version: v5.9.1 (latest)

### Summary / TL;DR:
- The accordion buttons that contain summary text in Swagger UI are named in a way that fails accessibility standards defined in [WCAG SC 2.5.3: Label in Name (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html)
- This error is present in every default installation of Swagger UI: https://github.com/swagger-api/swagger-ui/blob/master/src/core/components/operation-summary.jsx#L71
- For the websites I work on, that means thousands of accessibility violations across hundreds of our dataset pages 
- Removing the aria-label (editing a single line in the source) would spare us having to do a hacky workaround to eliminate the accessible name mismatch, and would benefit all users of Swagger UI, especially those who rely on assistive tech (voice control, screen readers, etc)

### Describe the bug you're encountering
The Swagger UI accordion button's accessible name (aria-label) does not include all of the text displayed on the button and therefore does not conform to accessibility standards defined in WCAG ([SC 2.5.3: Label in Name (Level A)](https://www.w3.org/TR/WCAG22/#label-in-name)). See also:

- [Understanding SC 2.5.3: Label in Name (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html)
- [axe-core rule: Elements must have their visible text as part of their accessible name](https://dequeuniversity.com/rules/axe/4.8/label-content-name-mismatch)

In the current button markup, the aria-label value is missing the summary description (e.g., "Add a new pet to the store"). Examples from the [Pet store demo](https://petstore.swagger.io/?_ga=2.114384135.1033731862.1698766740-1777610260.1698766740):

This markup fails SC 2.5.3:
```
<button aria-label="post /pet" aria-expanded="false" class="opblock-summary-control">
  <span class="opblock-summary-method">POST</span>
  <span class="opblock-summary-path" data-path="/pet">
    <a class="nostyle" href="#/pet/addPet"><span>/pet</span></a>
  </span>
  <div class="opblock-summary-description">Add a new pet to the store</div>
</button>
```

For comparison, this markup would pass SC 2.5.3 (aria-label removed):
```
<button aria-expanded="false" class="opblock-summary-control">
  <span class="opblock-summary-method">POST</span>
  <span class="opblock-summary-path" data-path="/pet">
    <a class="nostyle" href="#/pet/addPet"><span>/pet</span></a>
  </span>
  <div class="opblock-summary-description">Add a new pet to the store</div>
</button>
```

### To reproduce...
Some automated testing tools like Wave and axe DevTools do not flag this issue, however [Accessibility Insights](https://accessibilityinsights.io/downloads/) and Siteimprove do, as does manual testing with a screen reader.  Accessibility Insights flags the "label-content-name-mismatch" issue under "Needs review" for a person to verify the content mismatch.

Steps to reproduce: 
1. Open the [pet store demo](https://petstore.swagger.io/?_ga=2.114384135.1033731862.1698766740-1777610260.1698766740#/) (or other default installation of Swagger UI) in a browser with the Accessibility Insights extension installed
2. Open Accessibility Insights, and choose the FastPass option
3. When the results window appears, choose "(3) Needs review" from the left nav, and expand to show all instances of the "label-content-name-mismatch" issue.

### Expected behavior
There should be no label-content-name-mismatch occurrences displayed in Accessibility Insights, however there are 20 in the pet store demo. 

### Additional context or thoughts
Note that this issue applies only to the buttons that contain an **aria-label and visible text**. The arrow buttons for toggling accordions should retain their aria-label attributes because they do not have visible text. 

**Suggested fix:** The simplest way to correct this mismatch is to remove the aria-label from the accordion's summary button because it's entirely redundant. Without the aria-label, assistive tech will use the visible text to compute the accordion button's accessible name:

![image](https://github.com/swagger-api/swagger-ui/assets/228885/45d12646-887c-4616-a01b-7fb7599f1674)

**Comment by mgifford** [comment]
If the semantics can be achieved through HTML, we shouldn't be using ARIA. Redundant semantics doesn't benefit users.

**Comment by konstantintieber** [comment]
@mgifford so you are saying that rather than including the endpoint description in the existing aria-label attribute on that button ([link from the issue description](https://github.com/swagger-api/swagger-ui/blob/master/src/core/components/operation-summary.jsx#L71)), the proper and more sustainable way to improve accessibility would be to remove the `aria-label` attribute altogether?

**Comment by maggiewachs** [comment]
> @mgifford so you are saying that rather than including the endpoint description in the existing aria-label attribute on that button ([link from the issue description](https://github.com/swagger-api/swagger-ui/blob/master/src/core/components/operation-summary.jsx#L71)), the proper and more sustainable way to improve accessibility would be to remove the `aria-label` attribute altogether? Since all the text we include in the button is already contained within its children?

Yes! That's the simplest and most effective fix.

**Comment by char0n** [comment]
Hi @maggiewachs,

Thanks for explaining the whole context and reasoning.

**Comment by char0n** [comment]
Closed by https://github.com/swagger-api/swagger-ui/pull/9361
